fee/src/lib.rs

```diff
@@ -28,13 +28,140 @@ pub fn calculate_fee_details(
     if zero_fees_for_test {
         return FeeDetails::default();
     }
-    let signature_fee = message
-        .num_total_signatures()
-        .saturating_mul(lamports_per_signature);
+
+    calculate_fee_details_internal(
+        SignatureCounts::from(message),
+        lamports_per_signature,
+        prioritization_fee,
+        remove_rounding_in_fee_calculation,
+    )
+}
+
+/// This function has the actual fee calculation.
+fn calculate_fee_details_internal(
```

Comment:
I added this function as an additional internal level of abstraction.
It has only very simple argument types that are all explicitly typed. No traits hide the behavior; nearly any change to how the fee is calculated is immediately obvious from this code.

Not super satisfied with `prioritization_fee` being just passed in like it is now. Code for determining the prioritization fee should imo be inside this fee crate. But it's outside scope of this PR which is focusing on changing how we were using signature counts in the fee-calculation.

Reply:
For now can we just have an internal fn that calculates the signature fee? I don't love that this function otherwise just wraps the `FeeDetails` constructor and we have all these unit tests which are really only testing signature fees anyways.